### PR TITLE
[PS-1793] Desktop/Browser/Web: tweak disclosure widget design

### PR DESCRIPTION
**Pull request by patrickhlauke**
## Type of change

```
- [x] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Currently, there are disclosure widgets that use the `bwi-chevron-down` (when closed) and `bwi-angle-up` (when open), and have the icon positioned after their text.

Having the chevron/arrow on the right makes it look a bit "disheveled" ... particularly when there's multiple disclosure widgets, like on the options/settings in desktop/browser. it ends up with a very "ragged right" look as the various chevrons are obviously not vertically aligned, since the text labels vary in length.

screenshot showing three disclosure widgets in the browser settings/options ... all at different lengths, with the chevrons not vertically aligned

The up/down chevron icons look slightly confusing, and are not consistent with other disclosure widgets that use `bwi-angle-right` (when closed) and `bwi-angle-down` (when open) (e.g. in the left-hand column on desktop/web). These other disclosure widgets also have the icon positioned to the left of the text, not the right.

screenshot showing three disclosure widgets in the left-hand column of the desktop application, with the right-pointing chevron for closed and down-pointing chevron for opened disclosures, and the chevrons on the left all vertically aligned

This PR changes the former type of disclosure widget control to match the latter.

In addition, this PR makes a first tiny change to the web interface, to use `<button>` for the one disclosure widget (in "Send") it uses.

## Code changes

- **apps/browser/src/popup/send/send-add-edit.component.html**, **apps/browser/src/popup/settings/options.component.html**,  **apps/desktop/src/app/accounts/settings.component.html**, **apps/desktop/src/app/send/add-edit.component.html**: move the icon to the start of the `<button>`, change the `bwi-...` icons used
- **apps/browser/src/popup/scss/box.scss** and **apps/desktop/src/scss/box.scss**: change the margin from left to right to account for new icon position
- **apps/web/src/app/send/add-edit.component.html**: change the structure to `<h3><button>...</button></h3>` (to match changes made to desktop/browser apps ages ago), move the icon to the start of the `<button>`, change the `bwi-...` icons used
- **apps/web/src/scss/buttons.scss**: add new styles for the new button (since the web version doesn't use/have the same `.box .box-header-expanded` styles as desktop/browser
- **libs/components/src/link/link.stories.ts**: change the storybook examples to have icon before text, use correct `bwi-...` icon
- **apps/desktop/src/app/accounts/environment.component.html** and **apps/desktop/src/app/vault/generator.component.html**: harmonise the icons - replace the +/- square icons with the right/down facing chevrons. in addition, in the generator, remove the incorrect `appA11yTitle` which overrides the accessible name of the control (leading to a failure of WCAG 2.0 2.5.3 Label in Name)

## Screenshots

Before/after comparisons (where the "before" is the current master branch, NOT the latest stable release - incidentally, nice work on the browser style refresh, hoping to see this make its way to the desktop app and eventually the web)

### Desktop

Send before:

desktop-send-before

desktop-send-before2

Send after:

desktop-send-after

desktop-send-after2

---

Settings before:

desktop-settings-before

Settings after:

desktop-settings-after

---

Custom environment before:

closed custom environment with the + icon

open custom environment with the - icon

Custom environment after:

closed custom environment with the right-pointing chevron

open custom environment with the down-pointing chevron

---

Generator before:

closed options with the + icon

open options with the - icon

Generator after:

closed options  with the right-pointing chevron

open options  with the down-pointing chevron

### Browser

Send before:

browser-send-before

browser-send-before2

Send after:

browser-send-after

browser-send-after2

---

Options before:

browser-options-before

browser-options-after

### Web

Send before:

web-send-options-before

web-send-options-before2

Send after:

web-send-options-after

web-send-options-after2

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Review by danielleflinn — CHANGES_REQUESTED** [review]
Thank you for this contribution. Can you also address the expand/collapse button in the desktop generator dialog? Currently it uses different icons than these other instances while providing the same behavior.

**Comment by patrickhlauke** [comment]
> Can you also address the expand/collapse button in the desktop generator dialog?

Ah, that one slipped me by. Sure thing, will sort that tonight.

**Comment by patrickhlauke** [comment]
@danielleflinn done. also spotted the same +/- icon in the custom environment component, and changed that as well

**Comment by patrickhlauke** [comment]
@danielleflinn would you mind re-reviewing this to clear the "changes requested"?

**Review by danielleflinn — DISMISSED** [review]

**Comment by patrickhlauke** [comment]
thanks @danielleflinn

**Review by djsmith85 — CHANGES_REQUESTED** [review]
@patrickhlauke: Thank you again for your work on improving our clients.

Just had a quick search for `bwi-chevron-up` and I see at least 3 instances that would need to be checked:

- `apps\browser\src\popup\vault\vault-select.component.html`
- `apps\desktop\src\app\layout\account-switcher.component.html`
- `bitwarden_license\bit-web\src\app\organizations\manage\sso.component.html`

@danielleflinn Should we also update `libs\components\src\stories\icons.stories.mdx` with this PR?

I'll do a thorough review, once the changes have been made.

**Comment by patrickhlauke** [comment]
> Just had a quick search for `bwi-chevron-up` and I see at least 3 instances that would need to be checked:
> 
> * `apps\browser\src\popup\vault\vault-select.component.html`
> * `apps\desktop\src\app\layout\account-switcher.component.html`
> * `bitwarden_license\bit-web\src\app\organizations\manage\sso.component.html`

Will check again, but from memory those act more like `<select>` dropdowns, so there it makes sense to have the up/down chevron. This here is specifically for expand/collapse disclosure widgets

> @danielleflinn Should we also update `libs\components\src\stories\icons.stories.mdx` with this PR?

I'll have a quick look there as well, just in case.

**Review by danielleflinn — APPROVED** [review]

File: apps/web/src/app/send/add-edit.component.html [hunk 1]
```diff
@@ -141,14 +141,16 @@ <h3 class="mt-5">{{ "share" | i18n }}</h3>
...
         >
-          <h3 class="mb-0 mr-2">{{ "options" | i18n }}</h3>
-          <a class="mb-1" href="#" appStopClick role="button">
-            <i
-              class="bwi"
-              aria-hidden="true"
-              [ngClass]="{ 'bwi-angle-down': !showOptions, 'bwi-chevron-up': showOptions }"
-            ></i>
-          </a>
+          <h3 class="mb-0 mr-2">
```
Comment (djsmith85) [inline apps/web/src/app/send/add-edit.component.html:144]: The markup looks good and behaves correctly, but when focussed, it looks weird with the jagged outline:
image

**Review by djsmith85 — CHANGES_REQUESTED** [review]
@patrickhlauke Nice work on this and just one small thing I noticed while testing.

File: apps/web/src/app/send/add-edit.component.html (hunk 1 above)
Comment (patrickhlauke) [reply apps/web/src/app/send/add-edit.component.html:144]: @djsmith85 Fixed (with an admittedly ugly super-specific patch, but I can't quite work out where within the cascade the line height gets messed up).

screenshot showing the fixed focus outline on Options

**Review by djsmith85 — APPROVED** [review]
@patrickhlauke: Changes are looking good and are ready to get passed on to QA.

**Comment by djsmith85** [comment]
@patrickhlauke Received approval from QA. Thank you for your work on this and this will be included in the next release.

**Comment by patrickhlauke** [comment]
perfect, thank you @djsmith85 @danielleflinn

**Pull request merged**